common/src/test/scala/uk/ac/wellcome/test/fixtures/ElasticsearchFixtures.scala

```diff
@@ -48,22 +49,28 @@ trait ElasticsearchFixtures
   }
 
   def withLocalElasticsearchIndex[R](indexName: String, itemType: String)(
-    testWith: TestWith[String, R]) = {
+   testWith: TestWith[Future[String], R]): R = {
```

Comment:
This is a tricky one. While I can see where you're coming from, just returning `String` here (instead of a `Future`) has an impact on other code. It means we have to wait on futures in the fixture code, something you're "not supposed to do" with Scala, for various reasons including performance. I also think it might be confusing to use `eventually` in non-testcase code.

Finally, it encourages downstream tests to use for-comprehensions, which are safer especially with `Futures`. For example, the current `master` version of `ElasticSearchIndexTest` doesn't actually wait properly on one of the threads it spawns, leading to test instability:

https://github.com/wellcometrust/platform/blob/741cfcff27fcd8f8a16e4c62419fdac3f4c40aec/common/src/test/scala/uk/ac/wellcome/elasticsearch/ElasticSearchIndexTest.scala#L86

Thoughts?